configuration/pih/scripts/visit/encounterTypeConfig.js

```diff
@@ -1278,25 +1347,43 @@ angular.module("encounterTypeConfig", [])
         };
 
         return {
-          get: function(uuid, country, site) {
-            var encounterType = encounterTypes[uuid];
+          get: function(encounter, country, site) {
+
+            if (encounter == null || encounter.encounterType == null) {
+              return encounterTypes.DEFAULT;
+            }
+
+            let encounterType = encounterTypes[encounter.encounterType.uuid];
 
             if (encounterType == null) {
               return encounterTypes.DEFAULT;
             }
 
+            // resolve any country/site specific configs
             if (encounterType.hasOwnProperty(country)) {
-               if (encounterType[country].hasOwnProperty(site)) {
-                 return encounterType[country][site];
-               }
-               else {
-                 return encounterType[country].hasOwnProperty('DEFAULT') ?
-                   encounterType[country]['DEFAULT'] : encounterType[country];
-               }
+              if (encounterType[country].hasOwnProperty(site)) {
+                encounterType = encounterType[country][site];
+              } else {
+                encounterType = encounterType[country].hasOwnProperty('DEFAULT') ?
+                  encounterType[country]['DEFAULT'] : encounterType[country];
+              }
+            } else {
+              encounterType = encounterType.hasOwnProperty(['DEFAULT']) ?
+                encounterType['DEFAULT'] : encounterType;
+            }
```

Comment:
We can replace lines 1363-1373 with

```suggestion
              encounterType = encounterType[country][site] ||
                  encounterType[country]['DEFAULT'] ||
                  encounterType[country] ||
                  encounterType['DEFAULT'] ||
                  encounterType;
```

Reply:
Nice @brandones ! This is much cleaner and easier to read.

Reply:
Thanks @brandones !